lib/cn/csched_sp3.h

```diff
@@ -68,6 +69,9 @@ sp3_compact_request(struct csched *handle, int flags);
 void
 sp3_compact_status_get(struct csched *handle, struct hse_kvdb_compact_status *status);
 
+void
+sp3_dirty_node(struct csched *handle, struct cn_tree_node *tn);
+
```

Comment:
I don't care for this being a public interface.  To avoid a crash we need to check spn_initialized in sp3_dirty_node_locked() and bail if the spn is not initialized.  I think this is due to a delay in split node creation where a splill can find a node that has not yet been adopted by csched yet.  That's my theory anyway...

Ultimately the cleanest solution is to queue the work onto a csched queue and let csched manage things, like how it does now for compaction job completion.  Until then I can probably let this slide, but even then I think a callback/upcall provided by csched to the compaction job for this purpose is cleaner than exposing this as a public interface.

Reply:
Ah, I think this should work similar to sp3_process_new_trees(), but handle incrementally-spilled kvsets instead.  Lemme see if that can be cleanly implemented...

Reply:
I.e., we have cw_completion which is tied to sp3_work_complete().  We can have something similar for completed kvsets and then csched will just pick them up like it does other completed work...

Reply:
Nevermind, that requires post-spill synchronization before you can remove the root kvsets to be sure all the csched async work has been completed...